### PR TITLE
GEODE-7731: fix the checking of the end state of test.

### PR DESCRIPTION
**Pull request by echobravopapa**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/LocatorDUnitTest.java [hunk 1]
```diff
@@ -1314,17 +1314,21 @@ public void run() {
...
-      vm2.invoke(() -> {
-        GeodeAwaitility.await()
-            .until(() -> system.getDistributedMember().equals(getView().getCreator()));
-      });
-    }
+    await().untilAsserted(() -> {
+      MemberIdentifier viewCreator = vm1.invoke(() -> getView().getCreator());
+      MemberIdentifier viewCreator2 = vm1.invoke(() -> getView().getCreator());
+
+      if (isSystemConnected()) {
```
Comment (upthewaterspout) [inline geode-core/src/distributedTest/java/org/apache/geode/distributed/LocatorDUnitTest.java]: If isSystemConnected returns false, that means the test will pass immediately. Is that your intention?

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/LocatorDUnitTest.java [hunk 2]
```diff
@@ -1314,17 +1314,21 @@ public void run() {
     vm1.invoke("waitUntilLocatorBecomesCoordinator", this::waitUntilLocatorBecomesCoordinator);
```
Comment (bschuchardt) [inline geode-core/src/distributedTest/java/org/apache/geode/distributed/LocatorDUnitTest.java]: I think that waitUntilLocatorBecomesCoordinator should also use view.getCreator(), not view.getCoordinator().  This is allowing the test to proceed past these checks before a locator is actually coordinator - the view merely thinks that one of the locators should _become_ the coordinator.

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/LocatorDUnitTest.java (hunk 1 above)
Comment (echobravopapa) [reply geode-core/src/distributedTest/java/org/apache/geode/distributed/LocatorDUnitTest.java]: uh, nope... I was just guarding against the NPE that was happening...

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/LocatorDUnitTest.java (hunk 2 above)
Comment (echobravopapa) [reply geode-core/src/distributedTest/java/org/apache/geode/distributed/LocatorDUnitTest.java]: I'll have a look at that one too, thx

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/LocatorDUnitTest.java [hunk 3]
```diff
@@ -1314,17 +1319,20 @@ public void run() {
...
-      vm2.invoke(() -> {
-        GeodeAwaitility.await()
-            .until(() -> system.getDistributedMember().equals(getView().getCreator()));
-      });
-    }
+    await().untilAsserted(() -> {
+      MemberIdentifier viewCreator = vm1.invoke(() -> getView().getCreator());
+      MemberIdentifier viewCreator2 = vm1.invoke(() -> getView().getCreator());
+
+      // return (InternalDistributedMember) MembershipManagerHelper.getCreator(system);
```
Comment (upthewaterspout) [inline geode-core/src/distributedTest/java/org/apache/geode/distributed/LocatorDUnitTest.java]: Remove commented out code.

**Pull request merged**